ibis/backends/tests/test_vectorized_udf.py

```diff
@@ -41,16 +41,27 @@ def add_one(s):
     return s + 1
 
 
-def create_add_one_udf(result_formatter):
-    return elementwise(input_type=[dt.double], output_type=dt.double)(
-        _format_udf_return_type(add_one, result_formatter)
+def create_add_one_udf(result_formatter, id):
+    @elementwise(input_type=[dt.double], output_type=dt.double)
+    def add_one_legacy(s):
+        return result_formatter(add_one(s))
+
+    @ibis.udf.scalar.pandas
+    def add_one_udf(s: float) -> float:
+        return result_formatter(add_one(s))
+
+    yield param(add_one_legacy, id=f"add_one_legacy_{id}")
+    yield param(
+        add_one_udf,
+        marks=[pytest.mark.notimpl(["pandas", "dask"])],
+        id=f"add_one_modern_{id}",
     )
 
 
 add_one_udfs = [
-    create_add_one_udf(result_formatter=lambda v: v),  # pd.Series,
-    create_add_one_udf(result_formatter=lambda v: np.array(v)),  # np.array,
-    create_add_one_udf(result_formatter=lambda v: list(v)),  # list,
+    *create_add_one_udf(result_formatter=lambda v: v, id="series"),
+    *create_add_one_udf(result_formatter=lambda v: np.array(v), id="array"),
+    *create_add_one_udf(result_formatter=lambda v: list(v), id="list"),
```

Comment:
I tried to adapt the current tests to also test the new API to demonstrate that they meet the same need.

We should deprecate the old one at some point soon.